### PR TITLE
User authorization

### PR DESCRIPTION
**Pull request by sallypikachu**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class ReviewsController < ApplicationController
+  before_action :authorize_user, except: [:index, :show]
```
Comment (davidrf) [inline app/controllers/reviews_controller.rb]: change to use `Devise`'s `authenticate_user!`

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -27,4 +29,11 @@ def review_params
       :review_body
     )
   end
+
+  def authorize_user
+    if !user_signed_in?
+      flash.notice = "Log in before reviewing!"
+      redirect_to user_session_path
+    end
+  end
```
Comment (davidrf) [inline app/controllers/reviews_controller.rb]: Delete this method

File: app/controllers/venues_controller.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 class VenuesController < ApplicationController
+  before_action :authorize_user, except: [:index, :show]
```
Comment (davidrf) [inline app/controllers/venues_controller.rb]: use `authenticate_user!`

File: app/controllers/venues_controller.rb [hunk 4]
```diff
@@ -36,4 +38,11 @@ def venue_params
       :price_range
     )
   end
+
+  def authorize_user
+    if !user_signed_in?
+      flash.notice = "Log in before adding a new venue!"
+      redirect_to user_session_path
+    end
+  end
```
Comment (davidrf) [inline app/controllers/venues_controller.rb]: delete this method

File: spec/features/08_user_authorization_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+
+feature "authorization" do
+  let!(:venue) { FactoryGirl.create(:venue) }
+  let!(:review) { FactoryGirl.create(:review, venue_id: venue.id) }
```
Comment (davidrf) [inline spec/features/08_user_authorization_spec.rb]: `let!(:review) { FactoryGirl.create(:review, venue: venue) }`

File: app/views/venues/index.html.erb [hunk 6]
```diff
@@ -6,8 +6,14 @@
       <h1>Venues</h1>
     </div>
     <div class="floating-container">
-      <% @venues.each do |venue| %>
-    </h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+      <% if @venues.class == Venue::ActiveRecord_Relation %>
+        <% @venues.each do |venue| %>
+      </h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+        <% end %>
+        <% else %>
```
Comment (MLampa) [inline app/views/venues/index.html.erb]: Remove  lines 9-13

File: app/views/venues/index.html.erb [hunk 7]
```diff
@@ -6,8 +6,14 @@
       <h1>Venues</h1>
     </div>
     <div class="floating-container">
-      <% @venues.each do |venue| %>
-    </h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+      <% if @venues.class == Venue::ActiveRecord_Relation %>
+        <% @venues.each do |venue| %>
```
Comment (MLampa) [inline app/views/venues/index.html.erb]: Remove lines 9-10

File: app/views/venues/index.html.erb [hunk 8]
```diff
@@ -6,8 +6,14 @@
...
+      </h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+        <% end %>
+        <% else %>
+        <% @venues.each do |venue| %>
+      </h4><%= link_to venue.content, venue_path(venue) %></h4>
+        <% end %>
```
Comment (MLampa) [inline app/views/venues/index.html.erb]: Remove lines 13-16

File: app/views/venues/index.html.erb [hunk 9]
```diff
@@ -6,8 +6,14 @@
       <h1>Venues</h1>
     </div>
     <div class="floating-container">
-      <% @venues.each do |venue| %>
-    </h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+      <% if @venues.class == Venue::ActiveRecord_Relation %>
```
Comment (MLampa) [inline app/views/venues/index.html.erb]: Remove line 9

**Pull request merged**